Block List: Insert blocks after as default block

aduth opened the pull request:
Salvaged from #6452

This pull request seeks to normalize the behavior of `BlockListBlock`'s "Enter" key handling to insert the _default_ block, not explicitly the paragraph block.

**Testing instructions:**

There should be no change in behavior. Notably, pressing Enter while the Image block placeholder is selected should still insert a paragraph (which is the default block unless otherwise configured).

aduth committed:
Block List: Insert blocks after as default block

jorgefilipecosta reviewed (commented):
I noticed two bugs:
Insert 2 image placeholders. Select the first image press enter. The new paragraph is created after the second image. In master, it was created in the middle.
Add a columns block and add an image inside, press enter on the placeholder, the new paragraph is created outside of the columns.

I think both bugs are the result of the problem described in https://github.com/WordPress/gutenberg/pull/7870/files#r201515058.

aduth committed:
Block List: Correctly pass attributes and rootUID to insertDefaultBlock

aduth committed:
Testing: Refactor content retrieval utility to use data select

Avoids issues around focus transitions when moving between code and visual editor†, improves performance.

† Worth noting that these are legitimate issues, but ones which shouldn't be "covered" by a utility, but rather as part of its own suite of tests.

aduth committed:
Testing: Verify block boundary focus default append

aduth commented:
Nice catch @jorgefilipecosta . I've pushed a few updates which should fix this. I added a new end-to-end test case, though it doesn't cover much beyond testing the Enter/Delete behaviors while the block focus boundary (e.g. image placeholder) is selected, which is certainly better than what we have. Since I was having some issues implementing it with our current `getHTMLFromCodeEditor`, I also took additional steps to refactor this function to use `window.wp.data.select` instead.

jorgefilipecosta reviewed (approved):
Things look fine in my tests now 👍
